contracts/core/extensions/CoreIssuanceOrder.sol

```diff
@@ -123,6 +127,51 @@ contract CoreIssuanceOrder is
         );
     }
 
+    /**
+     * Cancel an issuance order
+     *
+     * @param  _addresses      [setAddress, makerAddress, makerToken, relayerToken]
+     * @param  _values         [quantity, makerTokenAmount, expiration, relayerTokenAmount, salt]
+     * @param  _cancelQuantity Quantity of set to be filled
+     */
+    function cancelOrder(
+        address[4] _addresses,
+        uint[5] _values,
+        uint _cancelQuantity
+    )
+        external
+        isPositiveQuantity(_cancelQuantity)
+    {
+        OrderLibrary.IssuanceOrder memory order = OrderLibrary.IssuanceOrder({
+            setAddress: _addresses[0],
+            quantity: _values[0],
+            makerAddress: _addresses[1],
+            makerToken: _addresses[2],
+            makerTokenAmount: _values[1],
+            expiration: _values[2],
+            relayerToken: _addresses[3],
+            relayerTokenAmount: _values[3],
+            salt: _values[4],
+            orderHash: OrderLibrary.generateOrderHash(
+                _addresses,
+                _values
+            )
+        });
+
+        // Make sure cancel order comes from maker
+        require(order.makerAddress == msg.sender);
+
+        // Verify order is valid and return amount to be cancelled
+        validateOrder(
```

Comment:
does this matter if the user is trying to cancel?

Reply:
Would say it does to avoid adding unnecessary data to storage, also this function will contain the openOrderAmount calculations in the next PR, so more of a placeholder.